src/Widget/WidgetInterface.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Bolt\Widget;
+
+use Symfony\Component\HttpFoundation\Response;
+
+interface WidgetInterface
+{
+    public function getName(): string;
+
+    public function getType(): string;
+
+    public function getTarget(): string;
+
+    public function getPriority(): int;
+
+    public function getTemplate(): string;
+
+    public function setResponse(?Response $response = null): self;
```

Comment:
why widget neeeds a response to be set?

Reply:
It might need the response for changing something in the Response. Most widgets will not need this, so i've moved this to a new `interface ResponseAware`.